cfgmgr/buffermgrdyn.cpp

```diff
@@ -455,10 +455,16 @@ void BufferMgrDynamic::recalculateSharedBufferPool()
     }
 }
 
-void BufferMgrDynamic::checkSharedBufferPoolSize()
+void BufferMgrDynamic::checkSharedBufferPoolSize(bool force_update_during_initialization = false)
 {
     // PortInitDone indicates all steps of port initialization has been done
     // Only after that does the buffer pool size update starts
+    if (!m_portInitDone && !force_update_during_initialization)
```

Comment:
@stephenxs , let's discuss offline about this change

Reply:
The logic:
- The buffer pool will be initialized only once before m_portInitDone being true.
- All the buffer pools should be configured before the initialization

This is achieved by the logic in `if (!m_portInitDone)` branch in function `checkSharedBufferPoolSize`.
According to the logic, the checkSharedBufferPoolSize is called only once before m_portInitDone being true. This means only the first call takes effect and all the following will be ignored if checkSharedBufferPoolSize is called multiple times before m_portInitDone being true.
There is a timer to ensure to all the buffer pools have been configured when checkSharedBufferPoolSize is called.  
However, if there is another call issued before the timer is timeout, it can result in all buffer pools have not been configured when the checkSharedBufferPoolSize is called. This can result in error logs during initialization.
So we want to avoid checkSharedBufferPoolSize being called from elsewhere before m_portInitDone and allow it to be called only by the timer only.
`force_update_during_initialization` is true when it is called from the timer and false elsewhere. This ensures only the timer can call it during the initialization.